packages/w3p/src/providers/wrapped/evm-wallet-connect.ts

```diff
@@ -0,0 +1,316 @@
+    return {
+      address: this.#address,
+      chainId: this.#chainId,
+      isConnected: this.isConnected,
+    }
+  }
+
+  async init(): Promise<void> {
+    this.#rawProvider = await UniversalProvider.init({
+      logger: 'info',
+      relayUrl: 'wss://relay.walletconnect.com',
+      projectId: this.#projectId,
+    })
+
+    await this.#checkForPersistedSession()
+
+    await this.#setListeners()
+
+    this.emit(PROVIDER_EVENT_BUS_EVENTS.Initiated, this.#defaultEventPayload)
+  }
+
+  async connect(): Promise<void> {
+    await this.#rawProvider.connect({
+      namespaces: {
+        eip155: {
+          methods: [
+            'wallet_switchEthereumChain',
+            'wallet_addEthereumChain',
+            'eth_requestAccounts',
+
+            'eth_sendTransaction',
+
+            'eth_sign',
+            'personal_sign',
+            'eth_signTypedData',
+          ],
+          chains: ['eip155:1'],
+          events: [
+            PROVIDER_EVENTS.Connect,
+            PROVIDER_EVENTS.Disconnect,
+            PROVIDER_EVENTS.ChainChanged,
+            PROVIDER_EVENTS.AccountsChanged,
+          ],
+        },
+      },
+      optionalNamespaces: {
+        eip155: {
+          methods: [
+            'wallet_switchEthereumChain',
+            'wallet_addEthereumChain',
+            'eth_requestAccounts',
+
+            'eth_sendTransaction',
+
+            'eth_sign',
+            'personal_sign',
+            'eth_signTypedData',
+          ],
+          chains: ['eip155:1'],
+          events: [
+            PROVIDER_EVENTS.Connect,
+            PROVIDER_EVENTS.Disconnect,
+            PROVIDER_EVENTS.ChainChanged,
+            PROVIDER_EVENTS.AccountsChanged,
+          ],
+        },
+      },
+      skipPairing: true,
+    })
+
+    await this.#rawProvider.enable()
+
+    this.#ethProvider = new providers.Web3Provider(this.#rawProvider)
+
+    const accounts = await this.#rawProvider.request<string[]>({
+      method: 'eth_requestAccounts',
+    })
+
+    this.#chainId =
+      this.#rawProvider?.session?.namespaces?.eip155?.chains?.[0]?.split(':')[1]
+
+    this.#address = accounts?.[0]
+
+    this.emit(
+      PROVIDER_EVENT_BUS_EVENTS.AccountChanged,
+      this.#defaultEventPayload,
+    )
+
+    this.emit(
+      this.isConnected
+        ? PROVIDER_EVENT_BUS_EVENTS.Connect
+        : PROVIDER_EVENT_BUS_EVENTS.Disconnect,
+      this.#defaultEventPayload,
+    )
+
+    this.#w3Modal.closeModal()
+  }
+
+  async disconnect() {
+    await this.#rawProvider.disconnect()
+    this.#chainId = undefined
+    this.#address = undefined
+    this.emit(PROVIDER_EVENT_BUS_EVENTS.Disconnect, this.#defaultEventPayload)
+  }
+
+  async #checkForPersistedSession() {
+    if (!this.#rawProvider) {
+      throw new ReferenceError('EthereumProvider is not initialized.')
+    }
+
+    if (!this.#rawProvider?.session) return
+
+    this.#chainId =
+      this.#rawProvider?.session?.namespaces?.eip155?.chains?.[0]?.split(':')[1]
```

Comment:
We should left comments here why do we getting values 4 the `chainId` and `address` with splitting values

```ts
const eip155 = this.#rawProvider?.session?.namespaces?.eip155
this.#chainId = eip155?.chains?.[0]?.split(':')[1]
this.#address = eip155?.accounts?.[0]?.split(':')?.[2]
```